libraries/vmmanager/fizzy/fizzy_vm_backend.cpp

```diff
@@ -0,0 +1,269 @@
+
+#include <fizzy/fizzy.h>
+
+#include <koinos/exception.hpp>
+
+#include <koinos/vmmanager/fizzy/exceptions.hpp>
+#include <koinos/vmmanager/fizzy/fizzy_vm_backend.hpp>
+
+#include <exception>
+#include <optional>
+#include <string>
+
+#define FIZZY_MAX_CALL_DEPTH   251
+
+namespace koinos::vmmanager::fizzy {
+
+/**
+ * Convert a pointer from inside the VM to a native pointer.
+ */
+char* resolve_ptr( FizzyInstance* fizzy_instance, uint32_t ptr, uint32_t size )
+{
+   KOINOS_ASSERT( fizzy_instance != nullptr, null_argument_exception, "fizzy_instance was unexpectedly null pointer" );
+   size_t mem_size = fizzy_get_instance_memory_size( fizzy_instance );
+   char* mem_data = (char *) fizzy_get_instance_memory_data( fizzy_instance );
+   KOINOS_ASSERT( mem_data != nullptr, fizzy_returned_null_exception, "fizzy_get_instance_memory_data() unexpectedly returned null pointer" );
+
+   if( ptr == mem_size )
+   {
+      if( size == 0 )
+         return mem_data + mem_size;
+   }
+   else if( ptr > mem_size )
+   {
+      return nullptr;
+   }
+
+   // How much memory exists between pointer and end of memory?
+   uint32_t mem_at_ptr = mem_size - ptr;
+   if( uint64_t(mem_at_ptr) < uint64_t(size) )
+      return nullptr;
+
+   return mem_data + ptr;
+}
+
+fizzy_vm_backend::fizzy_vm_backend() {}
+fizzy_vm_backend::~fizzy_vm_backend() {}
+
+std::string fizzy_vm_backend::backend_name()
+{
+   return "fizzy";
+}
+
+void fizzy_vm_backend::initialize()
+{
+}
+
+const char* fizzy_error_code_name(FizzyErrorCode code) noexcept
```

Comment:
Prefer `std::string` over C strings.

Reply:
This particular function's written in C style with the intent that we should maybe make an upstream PR to add it to the Fizzy C API.